### DISCUSSION TITLE
Explorer context menu items don't show up on Windows?

### DISCUSSION THREAD

**Discussion by adhamfarrag** [microsoft/vscode-discussions#698] [Extension Development QnA] [answered]
Hi, I'm working on [Nuxtr ](https://github.com/nuxtrdev/nuxtr-vscode) and we offer Explorer context menu items. Debugging a new version on Windows made us realize that Menu items don't show up on Windows. Works perfectly on Mac/Linux. 

Here's an example: 

```json
 {
 command": "nuxtr.directCreateComponent",
           "when": "(nuxtr.isNuxtProject && resourceFilename =~ /(components.*)/) || ( nuxtr.isNuxtProject && resourceDirname =~ /(components.*)/ && explorerResourceIsFolder)",
                    "group": "navigation"
                },
```

What could be reason why they don't work ? Is it configuration?

**Comment by burkeholland** [comment]
Are you using Insiders? I wonder if it could be related to the OS context issue https://github.com/microsoft/vscode/issues/184676

**Reply by adhamfarrag** [reply]
Hey @burkeholland, I'm very grateful for your work. I personally learnt a lot from you.

I tried on both, didn't work on stable too.

**Reply by gjsjohnmurray** [reply]
Do I understand correctly that the Explorer context menu you're referring to is the VS Code Explorer, not Windows Explorer?

Did you try using the `Developer: Inspect Context Keys` command to confirm that your when-clause is going to evaluate true?

**Reply by adhamfarrag** [reply]
@gjsjohnmurray I wanted to wait until the project is open sourced to have a better discussion. Here's the repo url https://github.com/nuxtrdev/nuxtr-vscode

Also, there was a misunderstanding regarding the mentioned example. The issue was related to context menu inside the editor with these [two items](https://github.com/nuxtrdev/nuxtr-vscode/blob/4951e12267b92f1f4ac597794842f75a1b829332/package.json#L367) CC: @burkeholland 

```JSON
 "editor/context": [
                {
                    "command": "nuxtr.createPageTemplate",
                    "when": " nuxtr.isNuxtProject &&  resourcePath =~ /pages/ ",
                    "group": "5_cutcopypaste@1"
                },
                {
                    "command": "nuxtr.createLayoutTemplate",
                    "when": " nuxtr.isNuxtProject &&  resourcePath =~ /layouts/ ",
                    "group": "5_cutcopypaste@1"
                }
            ]
```

Originally this is how it looked like from an example I saw somewhere, worked well on linux/macos but not on Windows.

```JSON
 "editor/context": [
                {
                    "command": "nuxtr.createPageTemplate",
                    "when": " nuxtr.isNuxtProject &&  resourcePath =~ /pages\\// ",
                    "group": "5_cutcopypaste"
                },
                {
                    "command": "nuxtr.createLayoutTemplate",
                    "when": " nuxtr.isNuxtProject &&  resourcePath =~ /layouts\\// ",
                    "group": "5_cutcopypaste"
                }
            ]
```

**Reply by isidorn** [reply]
The issue is probably in the usage of the `resourcePath` contextKey. Have you tried some other resource context keys as an alternative? Here's a useful list https://code.visualstudio.com/api/references/when-clause-contexts#available-context-keys

**Reply by olguzzar** [reply]
Hi @adhamfarrag, were you able to try out some other resource context keys?

**Reply by adhamfarrag** [reply, marked as answer]
@isidorn It works now with this contextKey. `resourcePath =~ /pages/`. [please, check package.json](https://github.com/nuxtrdev/nuxtr-vscode/blob/8952db7e823c62a58d998bf0178a738ab7eb2597/package.json#L371)

Hi @olguzzar, I managed to make it work using the example above.
